Print TE CUDA kernel

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42692 Print TE CUDA kernel**

Differential Revision: [D22986112](https://our.internmc.facebook.com/intern/diff/D22986112)

bertmaher committed:
Print TE CUDA kernel

[ghstack-poisoned]

zheng-xq commented on torch/csrc/jit/tensorexpr/cuda_codegen.cpp:
```diff
@@ -718,6 +717,18 @@ class NoThreadIdxRewriter : public IRMutator {
   bool need_rewrite_ = false;
 };
 
+std::ostream& operator<<(
```
I am not a fan of overloading "operator<<" for a generic data structure. I am perfectly fine if it is either static or within an anonymous namespace.

bertmaher replied:
I'll make this static, but I'm not sure I understand the objection?

zheng-xq commented on benchmarks/fastrnns/bench.py:
```diff
@@ -73,11 +74,15 @@ def train_batch(modeldef):
 
         # XXX: Use if need to print something
-        # print(modeldef.forward.graph_for(*modeldef.inputs))
+        global print_once
```
If I remember correctly, this function is called for different rnn_creator, so why a global "print_once"? I am perfectly fine if this flag is local to a particular training, and only once across all the training steps.

bertmaher replied:
Oops, this is not supposed to be here :)

ZolotukhinM commented on torch/csrc/jit/tensorexpr/cuda_codegen.cpp:
```diff
@@ -963,11 +957,6 @@ void CudaCodeGen::CompileToNVRTC(
   int major, minor;
   getMajorMinor(prop, major, minor);
 
-#if DEBUG_PRINT
```
Do we not need this in debug logs?

bertmaher replied:
Seems kind of low-value to me, unless someone really wants it I was inclined to remove it for the sake of spamming the console less.

ZolotukhinM commented on benchmarks/fastrnns/bench.py:
```diff
@@ -55,6 +55,7 @@ def elapsed_time(self, end_event):
         return end_event.time - self.time
 
 
+print_once = True
```
Changes to this file seem unrelated. Are they intentional?

bertmaher committed:
Update on "Print TE CUDA kernel"




[ghstack-poisoned]

bertmaher committed:
Update on "Print TE CUDA kernel"


Differential Revision: [D22986112](https://our.internmc.facebook.com/intern/diff/D22986112)

[ghstack-poisoned]
